smarts/core/renderer.py

```diff
@@ -60,91 +47,115 @@ def required_to(cls, thing):
+            ) as vshader_path, pkg_resources.path(
+                glsl, "unlit_shader.frag"
+            ) as fshader_path:
+                unlit_shader = Shader.load(
+                    Shader.SL_GLSL,
+                    vertex=str(vshader_path.absolute()),
+                    fragment=str(fshader_path.absolute()),
+                )
+                root_np.setShader(unlit_shader)
+            return root_np
 
-            gltf.patch_loader(self.loader)
-            self.setBackgroundColor(0, 0, 0, 1)
+        def render_node(self, sim_root: NodePath):
+            # Hack to prevent other SMARTS instances from also rendering
+            # when we call poll() here.
+            hidden = []
+            with self._render_lock:
+                for np in self.render.children:
+                    if np != sim_root and not np.isHidden():
+                        np.hide()
+                        hidden.append(np)
+                self.taskMgr.mgr.poll()
+                for np in hidden:
+                    np.show()
 
-            # Displayed framerate is misleading since we are not using a realtime clock
-            self.setFrameRateMeter(False)
 
-        except Exception as e:
-            # Known reasons for this failing:
-            raise RendererException(
-                f"Error in initializing framework for opening graphical display and creating scene graph. "
-                "A typical reason is display not found. Try running with different configurations of "
-                "`export DISPLAY=` using `:0`, `:1`... . If this does not work please consult "
-                "the documentation.\nException was: {e}"
-            ) from e
+except ImportError:
 
-    def destroy(self):
-        super().destroy()
-        self.__class__.__it__ = None
-
-    def __del__(self):
-        self.destroy()
-
-    def setup_sim_root(self, simid: str):
-        root_np = NodePath(simid)
-        with self._render_lock:
-            root_np.reparentTo(self.render)
-        with pkg_resources.path(
-            glsl, "unlit_shader.vert"
-        ) as vshader_path, pkg_resources.path(
-            glsl, "unlit_shader.frag"
-        ) as fshader_path:
-            unlit_shader = Shader.load(
-                Shader.SL_GLSL,
-                vertex=str(vshader_path.absolute()),
-                fragment=str(fshader_path.absolute()),
-            )
-            root_np.setShader(unlit_shader)
-        return root_np
-
-    def render_node(self, sim_root: NodePath):
-        # Hack to prevent other SMARTS instances from also rendering
-        # when we call poll() here.
-        hidden = []
-        with self._render_lock:
-            for np in self.render.children:
-                if np != sim_root and not np.isHidden():
-                    np.hide()
-                    hidden.append(np)
-            self.taskMgr.mgr.poll()
-            for np in hidden:
-                np.show()
+    class _ShowBaseInstance:
+        pass
```

Comment:
Then we can do this:
```suggestion
    class _ShowBaseInstance:
        def __init__():
           raise RendererException(f"You may not have installed the [render] dependencies required to render the camera sensor observations. "
                "Install them first using the command `pip install -e .[render]` at the source directory."
           )
```

Reply:
Yeah this looks more neat. Thanks.